sdk/src/evaluations/eas.ts

```diff
@@ -0,0 +1,70 @@
+import { Offchain, SchemaEncoder } from "@ethereum-attestation-service/eas-sdk";
+import { TypedDataSigner } from "@ethersproject/abstract-signer";
+import { ethers } from "ethers";
+
+import { DEFAULT_CHAIN_ID } from "../constants.js";
+import { InvalidOrMissingError } from "../types/errors.js";
+import { DuplicateEvaluation, EvaluationData } from "../types/evaluation.js";
+
+type EasEvaluatorConfig = {
+  address: string;
+  chainId: number;
+  signer: TypedDataSigner;
+};
+
+export default class EasEvaluator {
+  offChain: Offchain;
+  schemaEncoder: SchemaEncoder;
+  signer: TypedDataSigner;
+
+  constructor({
+    config = { chainId: DEFAULT_CHAIN_ID, address: "", signer: new ethers.VoidSigner("") },
+  }: {
+    config?: EasEvaluatorConfig;
+  }) {
+    this.offChain = new Offchain({ address: config.address, chainId: config.chainId, version: "0.26" });
+    this.schemaEncoder = new SchemaEncoder("uint256 chainId, address contract, uint256 claimId, string uri");
+    this.signer = config.signer;
+  }
+
+  signOfflineEvaluation = async (evaluation: EvaluationData) => {
+    let encodedData;
+    if (evaluation.type === "duplicate") {
+      const data = evaluation.evaluation as DuplicateEvaluation;
+      const realHypercert = data.realHypercert;
+
+      // Initialize SchemaEncoder with the schema string
+      // TODO validate schema values
+      encodedData = this.schemaEncoder.encodeData([
+        { name: "chainId", value: realHypercert.chainId as string, type: "uint256" },
+        { name: "contract", value: realHypercert.contract as string, type: "address" },
+        { name: "claimId", value: realHypercert.claimId as string, type: "uint256" },
+      ]);
+    }
```

Comment:
If you use an ADT, you should be able to add an `assertNever` here to ensure you're doing exhaustive type checking
https://github.com/hypercerts-org/hypercerts/blob/main/frontend/lib/common.ts#L39

Reply:
Added in ADT the Typescripty fashion. Open to refactoring it later but the current itteration is strong on typings I think and I'd like to get the functional flow down

Reply:
pushed